force-app/main/default/classes/DummyJSONCallout.cls

```diff
@@ -1,136 +1,188 @@
+   * The method is primarily designed to work in conjunction with callouts that fetch user details from the external system.
+   *
+   * Fields from JSON used to populate Salesforce Contact
+   * Fields: Email, Phone, Birthdate, MailingStreet, MailingCity, MailingPostalCode, MailingState, MailingCountry
+   *
+   * Documentation: https://dummyjson.com/docs/users
+   *
+   * @param String jsonResponse The JSON string from the external system that contains user details.
+   * @return Contact The Salesforce Contact object populated with details from the JSON string.
+   */
+  @TestVisible // Allows test class to see this method. Since it is private, it would not be visible otherwise.
+  private static Contact createContactFromJson(String jsonResponse) {
+    // Deserialize the JSON string into map of primitive data types.
+    Map<String, Object> jsonMap = (Map<String, Object>) JSON.deserializeUntyped(
+      jsonResponse
+    );
+    Map<String, Object> addrMap = (Map<String, Object>) jsonMap.get('address');
+    Contact con;
+    // Create a new contact from the JSON response.
+    // Set the contact fields from the JSON response.
+    con = new Contact(
+      Firstname = (String) jsonMap.get('firstName'),
+      LastName = (String) jsonMap.get('lastName'),
+      Email = (String) jsonMap.get('email'),
+      Phone = (String) jsonMap.get('phone'),
+      Birthdate = Date.valueOf((String) jsonMap.get('birthDate')),
+      // Deserialize the address from the JSON response.
+      // Set the address fields on the contact.
+      MailingStreet = (String) addrMap.get('address'),
+      MailingCity = (String) addrMap.get('city'),
+      MailingState = (String) addrMap.get('state'),
+      MailingPostalCode = (String) addrMap.get('postalCode'),
+      MailingCountry = (String) addrMap.get('country')
+    );
+    return con;
+  }
+
+  /*
+   * Question 3
+   * Sends Contact information to an external system https://dummyjson.com/users/add ('DummyJsonUser') to create or update a user based on the provided Salesforce Contact ID.
+   * The method prepares a JSON payload with the contact's information and sends it to the 'DummyJsonUser' add endpoint.
+   * If the callout is successful (HTTP status code between 200 and 299), the method updates the contact's 'DummyJSON_Last_Updated__c' field in Salesforce to the current time.
+   * Salesforce Contact should be used to generate the JSON payload
+   * Fields: Id, FirstName, LastName, Email, Phone
+   * This callout method should work in triggers
+   *
+   * Documentation: https://dummyjson.com/docs/users
+   *
+   * @param contactId The Salesforce Contact ID used to generate the JSON payload for the external system.
+   */
+ @future(callout=true)
+  public static void postCreateDummyJSONUser(String contactId) {
+    Contact con = [SELECT Id, DummyJSON_Id__c, DummyJSON_Last_Updated__c FROM Contact WHERE Id =: contactId WITH 
```

Comment:
The query isn't necessary at this point since we don't know if the http request is going to be successful yet.